src/kudu/rpc/messenger.cc

```diff
@@ -185,6 +189,28 @@ MessengerBuilder& MessengerBuilder::set_reuseport() {
   return *this;
 }
 
+MessengerBuilder& MessengerBuilder::set_send_buf(int send_buf) {
+  if (send_buf < kMinSockBuf) {
+    LOG(WARNING) << "Cannot set socket send buffer to "
+                 << send_buf
+                 << ". SO_SNDBUF will not be set.";
```

Comment:
We're saying that SO_SNDBUF will not be set but we're setting it to 0 anyway? Maybe we should set it to kMinSockBuf instead?

Reply:
If set to 0, when we actually check to set the sock opt, it'll be skipped, which is why it effectively won't be set.